### PR TITLE
Allow to store/resume open editors across Continue On transitions (fix #193704)

### LINKED ISSUE #193704: Allow to store/resume open editors across Continue On transitions
Split out from https://github.com/microsoft/vscode/issues/183449

### PR DESCRIPTION
**Pull request by bpasero**
fix #193704

### CODE REVIEW

**Comment by bpasero** [comment]
@joyceerhl this implements support to update editor state when the underlying storage changes. However, so far I was not able to figure out how to test roaming locally to ensure this actually works with how we store state today. As you can see, the change does not go into any of the editor input serialisers (yet) because it may just work 🤞

File: src/vs/workbench/common/component.ts [hunk 1]
```diff
@@ -39,6 +41,14 @@ export class Component extends Themable {
 		return this.memento.getMemento(scope, target);
 	}
 
+	protected reloadMememnto(scope: StorageScope): void {
```
Comment (joyceerhl) [inline src/vs/workbench/common/component.ts:44]:
```suggestion
	protected reloadMemento(scope: StorageScope): void {
```

File: src/vs/workbench/browser/parts/editor/editorParts.ts [hunk 2]
```diff
@@ -273,18 +269,75 @@ export class EditorParts extends MultiWindowParts<EditorPart> implements IEditor
...
 		}
 	}
 
 	get hasRestorableState(): boolean {
 		return this.parts.some(part => part.hasRestorableState);
 	}
 
+	private onDidChangeMementoState(e: IStorageValueChangeEvent): void {
+		if (e.external && e.scope === StorageScope.WORKSPACE) {
+			this.reloadMememnto(e.scope);
```
Comment (joyceerhl) [inline src/vs/workbench/browser/parts/editor/editorParts.ts:308]:
```suggestion
			this.reloadMemento(e.scope);
```

**Review by joyceerhl — COMMENTED** [review]
Awesome! I verified this PR will restore editor state with a local test in Code-OSS:
- Clone the same repo twice and open one clone in VS Code
- Use `Cloud Changes: Store Working Changes in Cloud` to store editor state in one clone
- Open the other cloned repo, the stored editor state is restored since both clones have the same identity

**Review by joaomoreno — APPROVED** [review]

**Pull request merged**